### DISCUSSION TITLE
Clarification on `network.intercept`

### DISCUSSION THREAD

**Discussion by decaffeinatedio** [rubycdp/ferrum#381] [General]
Am I correct in understanding that there can only be a single `network.intercept` block per context? That is my takeaway from the following in the readme:

> You used to call authorize method without block, but since it's implemented using request interception there could be a collision with another part of your code that also uses request interception, so that authorize allows the request while your code denies but it's too late. The block is mandatory now.

But I'm not asking exclusively about the authorize case -- will having two network.intercept blocks operating on the same context cause unexpected behavior? 

Thanks!

**Comment by route** [comment]
You can have many blocks only if you handle them properly. There are additional arguments passed to the block `index`, `total`:

```ruby
browser = Ferrum::Browser.new

browser.network.intercept

browser.on(:request) do |request, index, total|
  if request.match?(/bla-bla/)
    request.abort
  else
    # This check allows to find if there's next block that can handle request
    index + 1 < total ? next : request.continue
  end
end

browser.on(:request) do |request, index, total|
  if request.match?(/lorem/)
    request.respond(body: "Lorem ipsum")
  else
    index + 1 < total ? next : request.continue
  end
end

browser.go_to("https://www.google.com")
```

Though it's not documented I believe. I wanted to add some easier public API but didn't have time.
